sdk/azcore/response.go

```diff
@@ -129,6 +129,16 @@ func (r *Response) UnmarshalAsXML(v interface{}) error {
 	return err
 }
 
+// UnmarshalError will unmarshal the received service error payload along with the corresponding unmarshal error.
+// NOTE: this method is only meant to be called when handling a service error.
+func (r *Response) UnmarshalError(unmarshalErr error) error {
+	payload, err := r.payload()
+	if err != nil {
+		return err
+	}
+	return fmt.Errorf("raw service error: %s\n%w", payload, unmarshalErr)
```

Comment:
I wouldn't wrap `unmarshalErr` here since we don't need/want callers to test for explicit unmarshall errors.

Reply:
Didnt we also want include the unmarshal error for ourselves?

Reply:
We want to include it but not wrap it, so just a `%s` should do the trick.

Reply:
Done